Assets/Scripts/Interactables/CocktailMiniGame.cs

```diff
@@ -0,0 +1,374 @@
+
+                    if (rightArrow != null && rightArrow.GetComponent<Image>() != null)
+                        rightArrow.GetComponent<Image>().sprite = rightUIPressed; 
+                }
+                else
+                {
+                    if (leftArrow != null && leftArrow.GetComponent<Image>() != null)
+                        leftArrow.GetComponent<Image>().sprite = leftUIPressed;
+
+                    if (rightArrow != null && rightArrow.GetComponent<Image>() != null)
+                        rightArrow.GetComponent<Image>().sprite = rightUIUnpressed;  
+                }
+
+                // Wait for the defined switch speed
+                yield return new WaitForSeconds(arrowSwitchSpeed);
+
+                //debugging to see if left and right arrows are assigned and alternating
+                Debug.Log(leftArrow);  // Check if leftArrow is null
+                Debug.Log(rightArrow); // Check if rightArrow is null
+                Debug.Log("LeftArrowActive: " + leftArrowActive + ", Left Arrow: " + leftArrow.GetComponent<Image>().sprite.name + ", Right Arrow: " + rightArrow.GetComponent<Image>().sprite.name);
+            }
+        }
+
+        //check if player hit the btn at the right time, works in tandem with coroutine
+        private void DetectPlayerInput()
+        {
+           //putting quit option here to see if the player can quit by hitting Q
+            if (Input.GetKeyDown(Quit))
+            {
+                Debug.Log("Quit key pressed");
+                EndCocktailMiniGame(false);
+                return; // Ensure we stop further processing in this frame
+            }
+
+
+            //arrowSwitchSpeed contorls how quickly arrows alternate, adjust for difficulty
+            //perfectHitWindow defines how much time after the switch is considered a perfect hit to gain the score multiplier
+
+            // Detect input for the left arrow, is arrow active+did player hit btn
+            //uses the difference between the current time and the time the arrow switched (time % arrowswitch), used to calculate score
+
+            if (leftArrowActive && Input.GetKeyDown(leftKey))
+            {
+                float timeSinceSwitch = arrowSwitchSpeed - Mathf.Abs(Time.time % arrowSwitchSpeed);
+                CalculateScore(timeSinceSwitch);
+            }
+
+            // Detect input for the right arrow
+            else if (!leftArrowActive && Input.GetKeyDown(rightKey))
+            {
+                float timeSinceSwitch = arrowSwitchSpeed - Mathf.Abs(Time.time % arrowSwitchSpeed);
+                CalculateScore(timeSinceSwitch);
+            }
+        }
+
+        //multiplier increases based on timing accuracy
+        private void CalculateScore(float timeSinceSwitch) 
+        {
+
+            // Increase multiplier for a perfect hit
+            if (timeSinceSwitch <= perfectHitWindow)
+            {
+                scoreMultiplier += 0.5f; 
```

Comment:
These values should be tunable, rather than set in stone within a function. At bare minimum, move these into serialized values adjustable in editor, or we could look at the TunableHandler/similar ideas.